src/validators/JsonWebKey.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 - MATTR Limited
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BlsCurveName, JsonWebKey } from "src/types";
+
+export const assertPublicJwk = (jwk: JsonWebKey): boolean => {
+  // Performs common checks, but also checks to make sure a d value is not included
+  // If a d value is included then this is a private key not a public key
+  return checkCommonBlsJwkValues(jwk) && typeof jwk.d === "undefined";
```

Comment:
Will this work for G2 key too? Appears the check is G1 specific

Reply:
Thanks for catching this. I left the G1 stuff in there for now, but I'll add that in a separate commit and make this function generic to either BLS key